packages/@aws-cdk/toolkit-lib/lib/api/garbage-collection/garbage-collector.ts

```diff
@@ -178,6 +178,14 @@ interface GarbageCollectorProps {
    * @default true
    */
   readonly confirm?: boolean;
+
+  /**
+   * Non-CDK stack names or glob patterns to skip when encountering unauthorized access errors during garbage collection.
+   * You must explicitly specify non-CDK stack names - CDK stacks will be rejected to prevent accidental asset deletion.
```

Comment:
"CDK stacks will be rejected to prevent accidental asset deletion"

this reads like we are checking for the user whether or not stacks supplied here are CDK stacks.